modules/solid_mechanics/test/tests/dynamics/time_integration/direct_central_difference.i

```diff
@@ -0,0 +1,131 @@
+###########################################################
+# This is a simple test with a time-dependent problem
+# demonstrating the use of a central difference with a
+# direct calculation of acceleration.
+#
+# Testing that the first and second time derivatives
+# are calculated correctly using the Central Difference Direct
+# method
+###########################################################
+
+[GlobalParams]
+    displacements = 'disp_x disp_y'
+[]
+
+[Mesh]
+    type = GeneratedMesh
+    dim = 2
+    xmin = -1
+    xmax = 1
+    ymin = -1
+    ymax = 1
+    nx = 1
+    ny = 1
+[]
+
+[Variables]
+    [disp_x]
+    []
+    [disp_y]
+    []
+[]
+
+[Functions]
+    [forcing_fn]
+        type = PiecewiseLinear
+        x = '0.0 0.1 0.2    0.3  0.4    0.5  0.6'
+        y = '0.0 0.0 0.0025 0.01 0.0175 0.02 0.02'
+    []
+[]
+
+[Kernels]
+    [DynamicSolidMechanics]
+        displacements = 'disp_x disp_y'
+    []
+    [massmatrix]
+        type = MassMatrix
```

Comment:
Does the mass matrix object accept density defined as a material property? If not I suggest adding one.

Reply:
It does not. It inherits from from `Reaction.h`, and renames the rate param, to be density. It's a bit odd. So most likely the best way to do this would be to add an additional optional parameter that takes in density as a material property. But that would still be a bit weird.

Reply:
I would be fine with only keeping the material property parameter. In the input file, you can still assign plain numeric literals to that parameter.

Reply:
In `MassMatrix` the density parameter is defined like this `  params.renameParam("rate", "density", "Optional density for scaling the computed mass.");`. 
Where the original parameter `rate` is a `Real`. 
I'm not sure if I can change this input to a material property without also modifying `Reaction.h`.

Reply:
or just create a new object called something like `MatMassMatrix`...

Reply:
That would be the more straight forward option

Reply:
[23fce96](https://github.com/idaholab/moose/pull/28175/commits/23fce96798e42135935011643212a8d05882c9c3)

I took a crack at it here

Reply:
It looks good to me. My only complaint is that `MatMassMatrix` and `MassMatrix` have many duplicate code. Could you perhaps factor out a common base class called `MassMatrixBase` and let both derive from it?

Reply:
Tagging @lindsayad as it technically belongs to the framework.

Reply:
weigh one code duplication against another. how much does it realistically inherit from Reaction.h?

Reply:
[f32f1e7](https://github.com/idaholab/moose/pull/28175/commits/f32f1e751f63f51fb0763df97dd8fae602363d41)

I took a crack at it here